src/Base/PlacementPyImp.cpp

```diff
@@ -163,22 +163,33 @@ PyObject* PlacementPy::translate(PyObject * args)
     return move(args);
 }
 
-PyObject* PlacementPy::rotate(PyObject *args) {
-    PyObject *obj1, *obj2;
+PyObject* PlacementPy::rotate(PyObject *args, PyObject *kw) {
     double angle;
-    if (!PyArg_ParseTuple(args, "OOd", &obj1, &obj2, &angle))
+    bool comp = false;

```

Comment:
This should be an un-assigned bool, since the default value is given in the XML file - if you change it to `rotate(center, axis, angle, comp=False) -> None`

Reply:
I guess that's a matter of taste.

My taste is: C++ code should always rely on itself, never on anything else. If - for whatever reason - the XML assignment would fail, the C++ code would fail as well.

Reply:
Documentation: comp explicitly mentioned in the parameter list, although it's a keyword (only) argument.
Fixed with 2143a9cfade7df993d9aeb098c796c32ce55bcbe

Reply:
> his should be an un-assigned bool, since the default value is given in the XML file - if you change it to `rotate(center, axis, angle, comp=False) -> None` as I recommended

No, the <UserDocu> tag in the xml file only is used to generate the docstrings, there is no relation with the code implementation.

Reply:
> My understanding of "keyword=true", btw was that this just says the method takes keyword arguments?

Yes, the attribute Keyword on Methode elements "tells" to the generator which parameters the PyCFunction callback should take and sets the appropriate flags.